### PR TITLE
refactor: enable prettier and fix the styling

### PR DESCRIPTION
**Pull request by bfabio**
`prettier` needs to be configured in eslint with `"plugin:prettier/recommended"`,
as documented at
https://github.com/prettier/eslint-plugin-prettier#recommended-configuration
    
The previous config didn't really work and acted as if prettier was
disabled.
    
Also use `classnames` to avoid string concatenation. The linter
doesn't catch all the occurrences, so there may be places where
concatenation for classes is still there.

Fix https://github.com/italia/designers.italia.it/issues/742
Fix https://github.com/italia/designers.italia.it/issues/630
Fix https://github.com/italia/designers.italia.it/issues/848

### CODE REVIEW

File: gatsby-config.js [hunk 1]
```diff
@@ -1,40 +1,41 @@
...
-    image : "https://designers.italia.it/assets/icons/logo-it.png",
-    twitterImage : "https://designers.italia.it/assets/icons/logo-it.png",
-    twitterSite : "@DesignersITA",
-    twitterCreator : "Dipartimento per la trasformazione digitale e AGID",
-    themeColor: "#0066cc"
+    title: "Designers Italia",
+    author: "Dipartimento per la trasformazione digitale e AGID",
+    lang: "it",
+    description:
+      "Il punto di riferimento per la progettazione dei servizi pubblici digitali",
```
Comment (Fupete) [inline gatsby-config.js:28]: @bfabio this could be on row 27 after `:`?

Comment (bfabio) [reply gatsby-config.js:28]: @Fupete This is `prettier` acting on https://prettier.io/docs/en/options.html#print-width, that's expected and good

File: src/components/highlight-cards/highlight-cards.js [hunk 2]
```diff
@@ -19,88 +20,87 @@ function HighlightCards({
...
+    `${background ? ` bg-${background}` : ""}` +
+    `${nospace ? "" : "  py-5 py-lg-7"}` +
+    `${nopadtop ? "  no-pad-top" : ""}`;
 
-  const styles = 'highlight-cards'
-	+ `${background ? ` bg-${background}` : ''}`
-  + `${nospace ? '' : '  py-5 py-lg-7'}`
-  + `${nopadtop ? '  no-pad-top' : ''}`
+  let cardStyles = classNames("col-12 col-md-6 mb-3 mb-md-4", {
+    "col-lg-4": !col4,
```
Comment (bfabio) [inline src/components/highlight-cards/highlight-cards.js:30]: @Fupete btw, I kept this as it was, but it doesn't look right to begin with

Comment (Fupete) [reply src/components/highlight-cards/highlight-cards.js:30]: Seems right to me. I remember came from a classNames example. Are you sure?

Comment (bfabio) [reply src/components/highlight-cards/highlight-cards.js:30]: @Fupete this looks backwards, it should be `"col-lg-4"` is `col4` is truthy.

Comment (Fupete) [reply src/components/highlight-cards/highlight-cards.js:30]: They mean something different... col4 = 4 columns layout in our templates. col-lg-4 mean 4 bootstrap columns (of 12 total) largeness for each column = 3 columns layout. I know, weird but it works.

**Pull request merged**